pkg/cache/tas_cache.go

```diff
@@ -18,47 +18,94 @@ package cache
 
 import (
 	"maps"
+	"slices"
 	"sync"
 
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
 	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	utiltas "sigs.k8s.io/kueue/pkg/util/tas"
 )
 
-type TASCache struct {
+type tasCache struct {
 	sync.RWMutex
-	client  client.Client
-	flavors map[kueue.ResourceFlavorReference]*TASFlavorCache
+	client     client.Client
+	flavors    map[kueue.ResourceFlavorReference]flavorInformation
+	topologies map[kueue.TopologyReference]topologyInformation
+	cache      map[kueue.ResourceFlavorReference]*TASFlavorCache
 }
 
-func NewTASCache(client client.Client) TASCache {
-	return TASCache{
-		client:  client,
-		flavors: make(map[kueue.ResourceFlavorReference]*TASFlavorCache),
+func NewTASCache(client client.Client) tasCache {
+	return tasCache{
+		client:     client,
+		flavors:    make(map[kueue.ResourceFlavorReference]flavorInformation),
+		topologies: make(map[kueue.TopologyReference]topologyInformation),
+		cache:      make(map[kueue.ResourceFlavorReference]*TASFlavorCache),
 	}
 }
 
-func (t *TASCache) Get(name kueue.ResourceFlavorReference) *TASFlavorCache {
+func (t *tasCache) Get(name kueue.ResourceFlavorReference) *TASFlavorCache {
 	t.RLock()
 	defer t.RUnlock()
-	return t.flavors[name]
+	return t.cache[name]
 }
 
 // Clone returns a shallow copy of the map
-func (t *TASCache) Clone() map[kueue.ResourceFlavorReference]*TASFlavorCache {
+func (t *tasCache) Clone() map[kueue.ResourceFlavorReference]*TASFlavorCache {
 	t.RLock()
 	defer t.RUnlock()
-	return maps.Clone(t.flavors)
+	return maps.Clone(t.cache)
 }
 
-func (t *TASCache) Set(name kueue.ResourceFlavorReference, info *TASFlavorCache) {
+func (t *tasCache) AddFlavor(flavor *kueue.ResourceFlavor) {
 	t.Lock()
 	defer t.Unlock()
-	t.flavors[name] = info
+	name := kueue.ResourceFlavorReference(flavor.Name)
+	if _, ok := t.flavors[name]; !ok {
+		flavorInfo := flavorInformation{
+			TopologyName: *flavor.Spec.TopologyName,
+			NodeLabels:   maps.Clone(flavor.Spec.NodeLabels),
+			Tolerations:  slices.Clone(flavor.Spec.Tolerations),
+		}
+		t.flavors[name] = flavorInfo
+		if tInfo, ok := t.topologies[flavorInfo.TopologyName]; ok {
+			t.cache[name] = t.NewTASFlavorCache(tInfo, flavorInfo)
+		}
+	}
+}
+
+func (t *tasCache) AddTopology(topology *kueuealpha.Topology) {
+	t.Lock()
+	defer t.Unlock()
+	name := kueue.TopologyReference(topology.Name)
+	if _, ok := t.topologies[name]; !ok {
```

Comment:
same - what if we need to update this?

Reply:
The topology levels are immutable for now, see [here](https://github.com/kubernetes-sigs/kueue/blob/37cb149fbd3641cf58b91aa9ce6460f490b087c6/apis/kueue/v1alpha1/topology_types.go#L88). The user needs to re-create the topology.